clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -303,6 +303,73 @@ check_sane_control_flow()
         if (!run_checker(memrefs, false))
             return false;
     }
+#ifdef UNIX
+    // Negative test (PC discontinuity): Transition from instr to kernel_xfer event
+    // marker.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(1, 1),
+            gen_marker(1, TRACE_MARKER_TYPE_KERNEL_EVENT, 3),
+        };
+        if (!run_checker(
+                memrefs, true, 1, 2,
+                "Non-explicit control flow has no marker - Discontinuity between "
+                "instruction and kernel event marker",
+                "Failed to catch PC discontinuity for an instruction followed by "
+                "kernel xfer marker")) {
+            return false;
+        }
+    }
+    // Positive test: Transition from instr to kernel_xfer event marker. goes to the next
+    // instruction.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(1, 1),   gen_marker(1, TRACE_MARKER_TYPE_KERNEL_EVENT, 2),
+            gen_instr(1, 101), gen_marker(1, TRACE_MARKER_TYPE_KERNEL_XFER, 102),
+            gen_instr(1, 2),
+        };
+        if (!run_checker(memrefs, false)) {
+            return false;
+        }
+    }
+    // Positive test: We should skip the check if there is no instruction before the
+    // kernel event. i.e. The previous instr address is zero.
+    {
+        std::vector<memref_t> memrefs = {
+            gen_instr(1, 0),
```

Comment:
If you're checking for the case where there's no prev instr, then this get_instr shouldn't be there. The zero is an internal default, so better to not create an actual instr with that pc.